api/blueprints/drbCollection.py

```diff
@@ -448,11 +491,35 @@ def _buildPublications(editions):
 
     return opdsPubs
 
-def removeEditionsFromCollection(dbClient, collection):
+def removeEditionsFromCollection(dbClient, collection, editionIDs=None, workUUIDs=None):
 
     '''Deleting the rows of collection_editions that were in the original collection'''
 
-    dbClient.session.execute(COLLECTION_EDITIONS.delete().where(COLLECTION_EDITIONS.c.collection_id == collection.id))
+    #Delete every row of the collection
+    if editionIDs == None and workUUIDs == None:
+        dbClient.session.execute(COLLECTION_EDITIONS.delete()\
+            .where(COLLECTION_EDITIONS.c.collection_id == collection.id))
+    #Delete the rows that match the editionIDs
+    if editionIDs != None:
+        for eid in editionIDs:
+            dbClient.session.execute(COLLECTION_EDITIONS.delete()\
+                .where(COLLECTION_EDITIONS.c.edition_id == eid))
```

Comment:
You could pretty easily avoid the loop here I think:

```
dbClient.session.execute(COLLECTION_EDITIONS.delete().where(COLLECTION_EDITIONS.c.edition_id.in_(editionIDs))
```
But these will all get executed in a single transaction as is anyway, so probably not a huge deal!

Reply:
Good suggestion on simplifying the code.